pkg/controllers/rollout/controller.go

```diff
@@ -329,6 +352,11 @@ func pickBindingsToRoll(allBindings []*fleetv1beta1.ClusterResourceBinding, late
 	klog.V(2).InfoS("Calculated the max number of bindings to remove", "clusterResourcePlacement", klog.KObj(crp),
 		"maxUnavailableNumber", maxUnavailableNumber, "minAvailableNumber", minAvailableNumber,
 		"lowerBoundAvailableBindings", lowerBoundAvailableNumber, "maxNumberOfBindingsToRemove", maxNumberToRemove)
+
+	// we can still update the bindings that are failed to apply already regardless of the maxNumberToRemove
+	for i := 0; i < maxUnavailableNumber && i < len(applyFailedUpdateCandidates); i++ {
```

Comment:
we don't care about the maxUnavailableNumber, either? as they're already failed binding.

Reply:
yes, but we also don't want to blast it too fast. This is kind of a middle ground.